java/Nikiforov-Oleksandr/src/com/zoolatech/lecture6/tasks/_3/UserRepository.java

```diff
@@ -0,0 +1,25 @@
+package com.zoolatech.lecture6.tasks._3;
+
+import java.util.Optional;
+
+public class UserRepository {
+    public UserCache cache;
+    public UserTable table;
+
+    public UserRepository(UserCache cache, UserTable table) {
+        this.cache = cache;
+        this.table = table;
+    }
+
+    public String findUserEmail(String id) throws UserMissingException {
+        Optional<User> user = cache.findUser(id);
+        if (user.isPresent()) {
```

Comment:
try to get rid of `isPresent` calls with the help of Optional's `or..` and `map` methods